webrecorder/browsertrix
# Frontend archives -> teams migration

Resolves https://github.com/webrecorder/browsertrix-cloud/issues/423.
- Renames all user-facing "Archives" to "Teams"
- Allows users to choose team from dropdown, if they are a part of more than one
- Non-superadmin users are automatically redirected to their "default" team (first archive they've created or have been added to)

### Manual testing
1. Run app `yarn start`
2. Log in as normal user. Verify you're redirected to your default team crawls page
3. If your user is a part of more than one team, verify that you can select the team from the nav bar. If you only have one team, verify you see the name of the team in the nav bar.
4. Log out and log in as super admin. Verify you're not redirected to a default team page. Verify you can select a team or view all teams from the nav bar.
5. Click on "Browsertrix Cloud" on home page. Verify you're taken to the admin dashboard if superadmin, or your team crawls for regular users.

### Screenshots
**Nav bar menu for non-superadmin user in single team:**
<img width="261" alt="Screen Shot 2022-12-21 at 12 11 36 PM" src="https://user-images.githubusercontent.com/4672952/208999617-7ff7bb75-bf51-4abf-beb3-151ce93411e0.png">

**Nav bar for non-superadmin user in multiple teams:**
<img width="1146" alt="Screen Shot 2022-12-21 at 12 18 57 PM" src="https://user-images.githubusercontent.com/4672952/208999428-81ab3ae5-b4fc-4261-84f3-6285ebe7ec0a.png">

**Dropdown open:**
<img width="260" alt="Screen Shot 2022-12-21 at 12 13 51 PM" src="https://user-images.githubusercontent.com/4672952/208999495-8d053bc3-175f-498b-ae75-87be6c04ae8b.png">

**Nav bar for super admin:**
<img width="1121" alt="Screen Shot 2022-12-21 at 12 18 39 PM" src="https://user-images.githubusercontent.com/4672952/208999317-bf844800-2227-4eb9-a24f-46f7485cfb78.png">


### Potential improvements
Right now the frontend is determining the default team by retrieving archives and selecting the first created archive/team. It'd be nicer experience to return the archive/team ID from the [`GET /api/users/{id}`](https://dev.browsertrix.cloud/api/docs#/users/users_user_api_users__id__get) endpoint, to eliminate the need to wait for `GET /api/archives` to resolve before redirecting the user. This would also set the groundwork for letting users pick their own default team from user settings in the future.
